### PR TITLE
OCD-1877-SED text update

### PR DESCRIPTION
**Pull request by ashwinimore**
Review request:
@andlar 
@brianlindsy 
@kekey1

### CODE REVIEW

File: src/test/java/RESTAssuredTests/APICallsTrailingslashUpdate.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+				//given().get("https://chpl.ahrqstg.org/rest/developers").then().statusCode(200).log().all();
+				//given().get("https://chpl.ahrqstg.org/rest/developers/").then().statusCode(200).log().all();
+				//given().get("https://chpl.ahrqstg.org/rest/products").then().statusCode(200).log().all();
+				//given().get("https://chpl.ahrqstg.org/rest/products/").then().statusCode(200).log().all();
+				//given().get("https://chpl.ahrqstg.org/rest/atls").then().statusCode(200).log().all();
+				//given().get("https://chpl.ahrqstg.org/rest/atls/").then().statusCode(200).log().all();
+				//given().get("https://chpl.ahrqstg.org/rest/users").then().statusCode(200).log().all();
+				//given().get("https://chpl.ahrqstg.org/rest/users/").then().statusCode(200).log().all();
+						
+}
```
Comment (andlar) [inline src/test/java/RESTAssuredTests/APICallsTrailingslashUpdate.java]: Is this test actually running any tests?

File: src/test/java/pageObjects/listingDetails_page.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import stepDefinitions.hooks;
+
+
+public class listingDetails_page {
+	
+	private WebDriver driver;
+	private static WebElement element = null;
+	
+	
+	public static void load_listings_details_page(WebDriver driver) {
```
Comment (andlar) [inline src/test/java/pageObjects/listingDetails_page.java:22]: Can this method take another parameter that is the ID of the listing? Then you could use this single method in many places, where your Feature is written as something like `Given I am on the Details page for Listing "8481"`

File: src/test/java/pageObjects/listingDetails_page.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+			driver = hooks.driver;
+			driver.get("https://chpl.ahrqstg.org/#/product/8481");
+			WebDriverWait wait = new WebDriverWait(driver, 20);
+			driver.manage().window().maximize();
+			String actualTitle = driver.getTitle();
+			System.out.println("page title is : "+actualTitle);
```
Comment (andlar) [inline src/test/java/pageObjects/listingDetails_page.java]: These last two lines don't have anything to do with the stated reason for the method

File: src/test/java/pageObjects/listingDetails_page.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+		
+	}
+	
+	public static void sed_details_accordion_open(WebDriver driver){
+	
+		driver.get("https://chpl.ahrqdev.org/#/product/8511");	
+		WebDriverWait wait = new WebDriverWait(driver, 20);
+		wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("div.panel:nth-child(4) > div:nth-child(1) > a:nth-child(1)")));
+
```
Comment (andlar) [inline src/test/java/pageObjects/listingDetails_page.java]: The first lines here should be part of the method above where we get to the page. This method is there to "open the sed section of the accordion"; it should not be navigating anywhere.

File: src/test/java/pageObjects/listingDetails_page.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+		WebElement expanded1 = driver.findElement(By.cssSelector("div.panel:nth-child(4) > div:nth-child(1) > a:nth-child(1)"));
+		expanded1.click();
+	}
+	
+	public static void sed_details_no_sed_data(WebDriver driver){
+		
+		driver.get("https://chpl.ahrqdev.org/#/product/8511");	
+		WebDriverWait wait = new WebDriverWait(driver, 20);
+		wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("div.panel:nth-child(4) > div:nth-child(1) > a:nth-child(1)")));
+
```
Comment (andlar) [inline src/test/java/pageObjects/listingDetails_page.java]: The first lines here should be part of the method above where we get to the page. This method is there to "open the sed section of the accordion"; it should not be navigating anywhere.

File: src/test/java/stepDefinitions/SEDTweaksSteps.java [hunk 6]
```diff
@@ -75,68 +60,38 @@ public void i_click_open_SED_accordion() throws Throwable {
...
-		expanded1.click();
-		
-		 WebElement Table_1 = driver.findElement(By.xpath("//*[@id='mainContent']/div[2]/div[2]/div/div/div[3]/div[2]/div/span/span[4]/table/thead/tr/th[1]"));
-		 List<WebElement> Rows = Table_1.findElements(By.tagName("tr"));
-		 System.out.println("No. of rows: "+Rows.size());
-		 
-		  
+		WebElement Table_1 = listingDetails_page.SED_criteria_table(driver);
+		List<WebElement> Rows = Table_1.findElements(By.tagName("tr"));
+		System.out.println("No. of rows: "+Rows.size());
```
Comment (andlar) [inline src/test/java/stepDefinitions/SEDTweaksSteps.java]: Why print this?

File: src/test/java/stepDefinitions/SEDTweaksSteps.java [hunk 7]
```diff
@@ -75,68 +60,38 @@ public void i_click_open_SED_accordion() throws Throwable {
...
+				  
 		 //to print all the values inside the table
```
Comment (andlar) [inline src/test/java/stepDefinitions/SEDTweaksSteps.java]: Why print all the rows? The method is looking for criteria title/number; what does that have to do with `System.out`-ing a lot of lines?

File: src/test/resources/Features/SEDTweaks.feature [hunk 8]
```diff
@@ -15,10 +15,11 @@ Then the header name should read as 'Date SED Testing was Completed'
 Scenario: Verify SED download option and button title
 Then the download button title should read as 'Download SED Details'
 
+Scenario: Verify usability report title update
+Then usability report text should read as 'Full Usability Report'
+
```
Comment (andlar) [inline src/test/resources/Features/SEDTweaks.feature]: This scenario should be something like:
```
Scenario: Verify usability report title update
Given I am on the listing details page for Listing "8381"
When I open the SED details accordion
Then the usability report text should read as "full usability report"
```

**Review by andlar — CHANGES_REQUESTED** [review]

**Review by kekey1 — COMMENTED** [review]
Agree with Andrew's comments; I would approve after those items are addressed.

File: src/test/java/RESTAssuredTests/APICallsTrailingslashUpdate.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        .header("API-KEY", "***")
+        .header("content-type", "application/json")
+        .header("Authorization", "Bearer token")
+        .when()
+        .get("https://chpl.ahrqstg.org/rest/announcements/")
+        .then()
+        .log()
+        .all();
+        //Assert.assertEquals(statusCode, 200, "Correct status code returned");		
+		given().get("https://chpl.ahrqstg.org/rest/announcements/").then().statusCode(200).log().all();
```
Comment (kekey1) [inline src/test/java/RESTAssuredTests/APICallsTrailingslashUpdate.java]: Do you want to also test announcements without a / ? Looks like all the other calls below test both with and without /.

**Review by kekey1 — CHANGES_REQUESTED** [review]

File: src/test/java/RESTAssuredTests/APICallsTrailingslashUpdate.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+import static org.hamcrest.Matchers.*;
+
+import org.junit.BeforeClass;
+
+public class APICallsTrailingslashUpdate {
+	
+	
+	
+	@Test
+	public void testAPIcallswithoutlashes()
```
Comment (brianlindsy) [inline src/test/java/RESTAssuredTests/APICallsTrailingslashUpdate.java]: I think you mean 'testAPIcallswithoutslashes()' ?

**Review by kekey1 — APPROVED** [review]

**Review by brianlindsy — APPROVED** [review]

**Review by andlar — APPROVED** [review]

**Pull request merged**